lighthouse-core/test/report/html/renderer/category-renderer-test.js

```diff
@@ -114,23 +125,6 @@ describe('CategoryRenderer', () => {
     category.description = prevDesc;
   });
 
-  // TODO(phulce): revisit if top-level warnings approach is too noisy
-  it.skip('renders audits with warnings as failed', () => {
```

Comment:
do we want to render an audit with warnings as failed? or keep it in passed?

Reply:
good policy question!

Let's see. here's the possible warnings we have right now (that are placed into Passed)

![image](https://user-images.githubusercontent.com/39191/40320084-4342cce6-5cdf-11e8-974a-0e13b7423e62.png)
![image](https://user-images.githubusercontent.com/39191/40320116-61d7d976-5cdf-11e8-8343-b093cf8fb05f.png)
![image](https://user-images.githubusercontent.com/39191/40320118-63f2b2bc-5cdf-11e8-9985-50b447155769.png)

I see the argument for these being visible by default, however in the case of timing/decode, there's not much a user can do about them. Either way, to be visible by default, they'd have to be placed in "Failed" and either we keep the green checkmark or fail them. Both are pretty weird.

So I'm comfy with saying that warnings are orthogonal to passing/failing and default report visibility. wdyt?

Reply:
Yeah I don't think there's much anyone can do about them either, I'm fine to hide especially now that we've got Sentry tracking 'em for Httparchive runs 👍

Reply:
My suggestion in https://github.com/GoogleChrome/lighthouse/issues/5273#issuecomment-390777039 was to use warnings as a way to get attention to an issue without penalizing the score. So I would definitely prefer for the warning to be visible by default (not collapsed). I understand that warnings may also be used for things outside of the developer's control so collapsing may make more sense. Is this something that can be configured by the audit at runtime, rather than forcing all warnings to behave the same?

I'd also love to see different UI treatment for warnings; not a green check and not a red exclamation. Could it be a yellow/orange exclamation to distinguish it that something's amiss?

Reply:
Here're some ideas to visually distinguish the orthogonal warnings from audits.

![inlinewarning-ideas](https://user-images.githubusercontent.com/7561395/40394201-4ac3f1f4-5dd7-11e8-8130-4d9f0faf7d8b.png)

Reply:
Sg. broke this into it's own issue: https://github.com/GoogleChrome/lighthouse/issues/5327 ("make sure warnings within Passed Audits are visible")